torch/_dynamo/eval_frame.py

```diff
@@ -262,19 +262,61 @@ def innermost_fn(fn):
     return unaltered_fn
 
 
+# The config to restore to should dynamo compile / recompile when
+# executing from the compiled function's _TorchDynamoContext
+config_cache = threading.local()
+
+
+def _maybe_init_guarded_config_cache():
+    if not hasattr(config_cache, "saved_config_and_hash"):
+        config_cache.saved_config_and_hash: Optional[  # type: ignore[misc]
+            Tuple[Dict[str, Any], bytes]
```

Comment:
this is now `Tuple[Dict[str, Any], str]`

Reply:
I think NamedTuple would be good, it should be performant too. And it will make code more readable and more difficult to screw up.

Reply:
Actually, NamedTuple is very slow when using field accessors. Better to use an object. https://medium.com/@jacktator/dataclass-vs-namedtuple-vs-object-for-performance-optimization-in-python-691e234253b9